aero.minova.core.application.system.service/src/test/java/aero/minova/core/application/system/controller/XSqlProcedureControllerTest.java

```diff
@@ -0,0 +1,169 @@
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(null);
+			inputTable.addRow(inputRow);
+		}
+		SqlProcedureResult sqlres = new SqlProcedureResult();
+		sqlres.setOutputParameters(inputTable);
+		XSqlProcedureResult xRes = new XSqlProcedureResult("test", sqlres);
+		results.add(xRes);
+
+		sqlres.setOutputParameters(inputTable);
+		xRes = new XSqlProcedureResult("test3", sqlres);
+		results.add(xRes);
+
+		// In test2 steht der Value, den wir dann haben wollen.
+		Table testTable = new Table();
+		testTable.setName("xpTest");
+		testTable.addColumn(new Column("TestText", DataType.STRING));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(new Value("TestString", null));
+			testTable.addRow(inputRow);
+		}
+		sqlres.setOutputParameters(testTable);
+		xRes = new XSqlProcedureResult("test2", sqlres);
+		results.add(xRes);
+
+		// XTable zum Aufrufen der Methode
+		Row inputRow = new Row();
+		inputRow.addValue(new Value("TestText", "test2"));
+		List<Row> rows = new ArrayList<>();
+		rows.add(inputRow);
+		inputTable.setRows(rows);
+
+		XTable xtable = new XTable();
+		xtable.setId("TestTable");
+		xtable.setTable(inputTable);
+
+		Table resTable = testSubject.fillInDependencies(xtable, results);
+
+		assertThat(resTable.getRows().get(0).getValues().get(0).getStringValue()).isEqualTo("TestString");
+	}
+
+	@Test
+	public void testFindxSqlResultSetValid() {
+		List<XSqlProcedureResult> results = new ArrayList<>();
+
+		Table inputTable = new Table();
+		inputTable.setName("spTest");
+		inputTable.addColumn(new Column("TestText", DataType.STRING));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(null);
+			inputTable.addRow(inputRow);
+		}
+		SqlProcedureResult sqlres = new SqlProcedureResult();
+		sqlres.setOutputParameters(inputTable);
+		XSqlProcedureResult xRes = new XSqlProcedureResult("test", sqlres);
+		results.add(xRes);
+		xRes = new XSqlProcedureResult("test2", sqlres);
+		results.add(xRes);
+		xRes = new XSqlProcedureResult("test3", sqlres);
+		results.add(xRes);
+		XSqlProcedureResult testres = testSubject.findxSqlResultSet("test2", results);
+
+		assertThat(testres.getId()).isEqualTo("test2");
+	}
+
+	@Test
+	public void testFindxSqlResultSetInvalid() {
+		List<XSqlProcedureResult> results = new ArrayList<>();
+
+		Table inputTable = new Table();
+		inputTable.setName("spTest");
+		inputTable.addColumn(new Column("TestText", DataType.STRING));
+		{
+			Row inputRow = new Row();
+			inputRow.addValue(null);
+			inputTable.addRow(inputRow);
+		}
+		SqlProcedureResult sqlres = new SqlProcedureResult();
+		sqlres.setOutputParameters(inputTable);
+		XSqlProcedureResult xRes = new XSqlProcedureResult("test", sqlres);
+		results.add(xRes);
+		xRes = new XSqlProcedureResult("test2", sqlres);
+		results.add(xRes);
+		xRes = new XSqlProcedureResult("test3", sqlres);
+		results.add(xRes);
+
+		Throwable exception = assertThrows(RuntimeException.class, () -> testSubject.findxSqlResultSet("test100", results));
```

Comment:
Ich empfehle folgenden Stil, dann braucht man auch keine Rule:
```
import static org.assertj.core.api.Assertions.catchThrowable;
[...]
catchThrowable(() -> testSubject.findxSqlResultSet("test100", results))
  .isInstanceOf(RuntimeException.class)
  .hasMessage("Cannot find SqlProcedureResult with Id test100");
```

Reply:
gibt bei mir einen Fehler: The method isInstanceOf(Class<RuntimeException>) is undefined for the type Throwable